data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/CommonTopicConfig.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+
+import java.time.Duration;
+
+/**
+ * This class has topic configurations which are common to all Kafka plugins - source, buffer, and sink.
+ * <p>
+ * Be sure to only add to this configuration if the setting is applicable for all three types.
+ */
+public abstract class CommonTopicConfig implements TopicConfig {
+    static final Duration DEFAULT_SESSION_TIMEOUT = Duration.ofSeconds(45);
+    static final String DEFAULT_AUTO_OFFSET_RESET = "earliest";
+    static final Duration DEFAULT_THREAD_WAITING_TIME = Duration.ofSeconds(5);
+    static final Duration DEFAULT_RETRY_BACKOFF = Duration.ofSeconds(10);
+    static final Duration DEFAULT_RECONNECT_BACKOFF = Duration.ofSeconds(10);
+    static final Duration DEFAULT_MAX_POLL_INTERVAL = Duration.ofSeconds(300);
```

Comment:
MAX_POLL_INTERVAL also doesn't belong in the common. common should be intersection of Producer and Consumer. We can get this info from https://docs.confluent.io/platform/current/installation/configuration/producer-configs.html and https://docs.confluent.io/platform/current/installation/configuration/consumer-configs.html.

Reply:
This is the common list between the two. At the topic level it looks like only the following two are common?
```
reconnect.backoff.ms
retry.backoff.ms
```

Things like bootstrap servers, auth (ssl, sasl, oauth configs) are common between source and sink. But the at topic level very few are common.